### PR TITLE
Added a check for property bag in tag helper descriptor comparer.

### PR DESCRIPTION
**Pull request by cjqian**
aspnet/mvc#1051

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorComparer.cs [hunk 1]
```diff
@@ -61,7 +62,8 @@ public virtual bool Equals(TagHelperDescriptor descriptorX, TagHelperDescriptor
                     descriptorX.AllowedChildren.OrderBy(child => child, StringComparer.OrdinalIgnoreCase),
                     descriptorY.AllowedChildren.OrderBy(child => child, StringComparer.OrdinalIgnoreCase),
                     StringComparer.OrdinalIgnoreCase))) &&
-                descriptorX.TagStructure == descriptorY.TagStructure;
+                descriptorX.TagStructure == descriptorY.TagStructure &&
+                descriptorX.PropertyBag.OrderBy(x => x.Key).SequenceEqual(descriptorY.PropertyBag.OrderBy(y => y.Key));
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorComparer.cs]: - Do Enumerable.SequenceEqual to stay consistent with the above.
- Provide Comparer in both order bys
- `propertyX`, `propertyY`

**Comment by NTaylorMullen** [comment]
:watch:

**Comment by cjqian** [comment]
:up: :date:

File: src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorComparer.cs [hunk 2]
```diff
@@ -61,7 +62,11 @@ public virtual bool Equals(TagHelperDescriptor descriptorX, TagHelperDescriptor
                     descriptorX.AllowedChildren.OrderBy(child => child, StringComparer.OrdinalIgnoreCase),
                     descriptorY.AllowedChildren.OrderBy(child => child, StringComparer.OrdinalIgnoreCase),
                     StringComparer.OrdinalIgnoreCase))) &&
-                descriptorX.TagStructure == descriptorY.TagStructure;
+                descriptorX.TagStructure == descriptorY.TagStructure &&
+                Enumerable.SequenceEqual(
+                    descriptorX.PropertyBag.OrderBy(propertyX => propertyX.Key, StringComparer.OrdinalIgnoreCase),
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorComparer.cs]: The `PropertyBag` was constructed with a `StringComparer.Ordinal` why would this differ?

File: src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorComparer.cs [hunk 3]
```diff
@@ -61,7 +62,11 @@ public virtual bool Equals(TagHelperDescriptor descriptorX, TagHelperDescriptor
...
+                    descriptorY.PropertyBag.OrderBy(propertyY => propertyY.Key, StringComparer.OrdinalIgnoreCase)
+                    );
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorComparer.cs]: nit: bring up to end of previous line

**Comment by NTaylorMullen** [comment]
:watch:

**Comment by cjqian** [comment]
:up: :date:

File: src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorComparer.cs [hunk 4]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections;
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorComparer.cs]: Prob don't need this anymore

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request merged**